### ISSUE TITLE
Virtual host weirdness with http -> https redirects

### ISSUE THREAD

**Issue by Doridian** [h2o/h2o#293]
My config currently looks like this, however it causes h2o to redirect even https -> https (for the non-www TLD. The www is always supposed to be stripped as seen by the lower 2 hosts).

Is this just a mistake on my part misunderstanding the configuration file or is this not supported yet?

```
proxy.preserve-host: ON
proxy.timeout.io: 60000
error-log: /var/log/h2o/error.log
access-log: /var/log/h2o/access.log
pid-file: /var/run/h2o.pid
user: h2o
listen:
  port: 80
  host: 0.0.0.0
listen:
  port: 443
  host: 0.0.0.0
  ssl:
    certificate-file: /etc/h2o/ssl/domain.tld.crt
    key-file: /etc/h2o/ssl/domain.tld.key
hosts:
  "default:80":
    paths:
      /:
        proxy.reverse.url: http://127.0.0.1:8080
  "default:443":
    paths:
      /:
        proxy.reverse.url: http://127.0.0.1:8080
  "domain.tld:80":
    paths:
      /:
        file.dir: /dev/null
        redirect: https://domain.tld
  "www.domain.tld:80":
    paths:
      /:
        file.dir: /dev/null
        redirect: https://domain.tld
  "www.domain.tld:443":
    paths:
      /:
        file.dir: /dev/null
        redirect: https://domain.tld
```

**Comment by kazuho** [comment]
Thank you for reporting the issue.  We would need to make changes to how H2O determines the host context corresponding to the requests received.

Please wait a while until we fix the issue.

**Comment by kazuho** [comment]
@Doridian 
I believe that the issue has been fixed in #296.  Would you mind checking?  Thank you in advance.
